tests/Credentials/InstanceProfileProviderTest.php

```diff
@@ -1204,4 +1205,154 @@ public function testResetsAttempts()
         $provider()->wait();
         $this->assertLessThanOrEqual(3, $this->getPropertyValue($provider, 'attempts'));
     }
+
+    /**
+     * This test checks for disabling IMDSv1 fallback by explicit client config passing.
+     *
+     * @return void
+     */
+    public function testIMDSv1DisabledByExplicitConfig() {
+        $config = [InstanceProfileProvider::CFG_EC2_METADATA_V1_DISABLED => true];
+        $wereCredentialsFetched = $this->fetchMockedCredentialsAndAlwaysExpectAToken($config);
+
+        $this->assertTrue($wereCredentialsFetched);
+    }
+
+    /**
+     * This test checks for disabling IMDSv1 fallback by setting AWS_EC2_METADATA_V1_DISABLED to true.
+     *
+     * @return void
+     */
+    public function testIMDSv1DisabledByEnvironment() {
+        $ec2MetadataV1Disabled = ConfigurationResolver::env(InstanceProfileProvider::CFG_EC2_METADATA_V1_DISABLED, 'string');
+        $putIMDSv1VarValue = function ($value) {
```

Comment:
let's just stick with putenv and clearenv (see other test classes) for setting and clearing the environment.